packages/fetch-engine/src/download.ts

```diff
@@ -291,19 +291,8 @@ async function binaryNeedsToBeDownloaded(
     }
   }
 
-  // If there is no cache and the file doesn't exist, we for sure need to download it
-  if (!targetExists) {
-    debug(`file ${job.targetFilePath} does not exist and must be downloaded`)
-    return true
-  }
-
-  // 3. If same platform, always check --version
-  if (job.binaryTarget === nativePlatform && job.binaryName !== BinaryType.libqueryEngine) {
-    const works = await checkVersionCommand(job.targetFilePath)
-    return !works
-  } // TODO: this is probably not useful anymore
-
-  return false
+  // if it's not cached, it needs to be downloaded
+  return true
```

Comment:
> That's a problem because it means we use whatever engine is available if one exists.
> Fortunately, this only affects us during local development and not actual users.

So the solution is to just download if we haven't cached it. If it is cached, it's served from there.